src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -633,7 +633,7 @@ void init(int opmode, Key key, AlgorithmParameterSpec params,
                 lastEncKey = keyBytes;
             }
 
-            if (useNativeGCM) {
+            if (useNativeGCM && NativeCrypto.isAllowedAndLoaded()) {
```

Comment:
Only an `instanceof` test is needed here:
```java
            if (cipher instanceof NativeGaloisCounterMode) {
```

Reply:
Also fixed.